src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/RewindDataIndexRepository.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.DataTypes;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Persistent implementation of the <see cref="RewindData"/> DBreeze repository.
+    /// </summary>
+    public class RewindDataIndexRepository : IRewindDataIndexRepository
+    {
+        /// <summary>
+        /// Instance logger.
+        /// </summary>
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Access to DBreeze database.
+        /// </summary>
+        private readonly DBreezeEngine dbreeze;
+
+        /// <summary>
+        /// DBreeze table names.
+        /// </summary>
+        private const string RewindDataIndexTable = "RewindDataIndex";
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="dBreezeCoinView"><see cref="DBreezeCoinView"/>.</param>
+        /// <param name="loggerFactory">Factory to create a logger for this type.</param>
+        public RewindDataIndexRepository(DBreezeCoinView dBreezeCoinView, ILoggerFactory loggerFactory)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.dbreeze = dBreezeCoinView.DBreeze;
+        }
+
+        /// <inheritdoc />
+        public Task<int?> GetAsync(string key, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Task<int?> task = Task.Run(() =>
+            {
+                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
```

Comment:
as said above, probably would be better to use something like `dBreezeCoinView.CreateDBreezeTransaction()` so we don't expose DBreeze engine (we should anyway move to a IDBreezeTransactionFactory service)

Reply:
Fixed